src/backend/distributed/commands/type.c

```diff
@@ -0,0 +1,1366 @@
+						  ENABLE_DDL_PROPAGATION);
+
+	return NodeDDLTaskList(ALL_WORKERS, commands);
+}
+
+
+/*
+ * PlanAlterTypeSchemaStmt is executed before the statement is applied to the local
+ * postgres instance.
+ *
+ * In this stage we can prepare the commands that need to be run on all workers.
+ */
+List *
+PlanAlterTypeSchemaStmt(AlterObjectSchemaStmt *stmt, const char *queryString)
+{
+	const char *sql = NULL;
+	const ObjectAddress *typeAddress = NULL;
+	List *commands = NIL;
+
+	Assert(stmt->objectType == OBJECT_TYPE);
+
+	if (creating_extension)
+	{
+		/* types from extensions are managed by extensions, skipping */
+		return NIL;
+	}
+
+	typeAddress = GetObjectAddressFromParseTree((Node *) stmt, false);
+	if (!IsObjectDistributed(typeAddress))
+	{
+		/* not distributed to the workers, nothing to do */
+		return NIL;
+	}
+
+	EnsureCoordinator();
+
+	QualifyTreeNode((Node *) stmt);
+	sql = DeparseTreeNode((Node *) stmt);
+
+	EnsureSequentialModeForTypeDDL();
+
+	commands = list_make3(DISABLE_DDL_PROPAGATION,
+						  (void *) sql,
+						  ENABLE_DDL_PROPAGATION);
+
+	return NodeDDLTaskList(ALL_WORKERS, commands);
+}
+
+
+/*
+ * ProcessAlterTypeSchemaStmt is executed after the change has been applied locally, we
+ * can now use the new dependencies of the type to ensure all its dependencies exist on
+ * the workers before we apply the commands remotely.
+ */
+void
+ProcessAlterTypeSchemaStmt(AlterObjectSchemaStmt *stmt, const char *queryString)
+{
+	const ObjectAddress *typeAddress = NULL;
+
+	Assert(stmt->objectType == OBJECT_TYPE);
+
+	if (creating_extension)
+	{
+		/* types from extensions are managed by extensions, skipping */
+		return;
+	}
+
+	typeAddress = GetObjectAddressFromParseTree((Node *) stmt, false);
+	if (!IsObjectDistributed(typeAddress))
+	{
+		/* not distributed to the workers, nothing to do */
+		return;
+	}
+
+	/* dependencies have changed (schema) lets ensure they exist */
+	EnsureDependenciesExistsOnAllNodes(typeAddress);
+}
+
+
+/*
+ * PlanAlterTypeOwnerStmt is called for change of owner ship of types before the owner
+ * ship is changed on the local instance.
+ *
+ * If the type for which the owner is changed is distributed we execute the change on all
+ * the workers to keep the type in sync across the cluster.
+ */
+List *
+PlanAlterTypeOwnerStmt(AlterOwnerStmt *stmt, const char *queryString)
+{
+	const ObjectAddress *typeAddress = NULL;
+	const char *sql = NULL;
+	List *commands = NULL;
+
+	Assert(stmt->objectType == OBJECT_TYPE);
+
+	if (creating_extension)
+	{
+		/* types from extensions are managed by extensions, skipping */
+		return NIL;
+	}
+
+	typeAddress = GetObjectAddressFromParseTree((Node *) stmt, false);
+	if (!IsObjectDistributed(typeAddress))
+	{
+		return NIL;
+	}
+
+	EnsureCoordinator();
+
+	QualifyTreeNode((Node *) stmt);
+	sql = DeparseTreeNode((Node *) stmt);
+
+	EnsureSequentialModeForTypeDDL();
+	commands = list_make3(DISABLE_DDL_PROPAGATION,
+						  (void *) sql,
+						  ENABLE_DDL_PROPAGATION);
```

Comment:
Minor: Maybe turn this into a function.

Reply:
IMHO it will be less readable if we abstract this to a function.
Now it is very visible we disable and enable ddl propagation around the remote command to be applied. I think that will be less clear if we hide it in a function